tests/test_ioc_system.py

```diff
@@ -178,144 +107,44 @@ async def test_create_softioc_update(
 #         purge_channel_caches()
 
 
-@pytest.mark.asyncio
-async def test_create_softioc_record_update_send_to_panda(
-    dummy_server_system: DummyServer,
-    subprocess_ioc,
-):
-    """Test that updating a record causes the new value to be sent to PandA"""
-    # Set the special response for the server
-    dummy_server_system.expected_message_responses.update(
-        {"PCAP1.TRIG_EDGE=Either": "OK"}
-    )
-
-    await caput(TEST_PREFIX + ":PCAP1:TRIG_EDGE", "Either", wait=True, timeout=TIMEOUT)
-
-    # Confirm the server received the expected string
-    assert (
-        "PCAP1.TRIG_EDGE=Either" not in dummy_server_system.expected_message_responses
-    )
-
-
-@pytest.mark.asyncio
-async def test_create_softioc_arm_disarm(
-    dummy_server_system: DummyServer,
-    subprocess_ioc,
-):
-    """Test that the Arm and Disarm commands are correctly sent to PandA"""
-
-    # Set the special response for the server
-    dummy_server_system.expected_message_responses.update(
-        {"*PCAP.ARM=": "OK", "*PCAP.DISARM=": "OK"}
-    )
-
-    await caput(TEST_PREFIX + ":PCAP:ARM", 1, wait=True, timeout=TIMEOUT)
-
-    await caput(TEST_PREFIX + ":PCAP:ARM", 0, wait=True, timeout=TIMEOUT)
-
-    # Confirm the server received the expected strings
-    assert "*PCAP.ARM=" not in dummy_server_system.expected_message_responses
-    assert "*PCAP.DISARM=" not in dummy_server_system.expected_message_responses
-
-
 def test_ensure_block_number_present():
     assert _ensure_block_number_present("ABC.DEF.GHI") == "ABC1.DEF.GHI"
     assert _ensure_block_number_present("JKL1.MNOP") == "JKL1.MNOP"
 
 
-@pytest.mark.asyncio
-async def test_create_softioc_time_panda_changes(
```

Comment:
The contents of this test are copied in full into the next test 

``` python
async def test_create_softioc_time_epics_changes()
```

In the removed first test there was
```
        assert await asyncio.wait_for(egu_queue.get(), TIMEOUT) == "s"
        assert await asyncio.wait_for(units_queue.get(), TIMEOUT) == "s"
        assert await asyncio.wait_for(drvl_queue.get(), TIMEOUT) == 8e-09
```

But in my test environment my `drvl` ends up as `1.333333333e-10`, consistent with a conversion from `ms` to `min`, even though I'm converting from `ms` to `sec` (and the units field correctly makes this change). I'm trying to find out if this is an error in my test, or in `update_drvl()`, since my test client doesn't  ever convert  to minutes through the `MockedAsyncioClient` changes.

In the `test_create_softioc_time_epics_changes()` we convert to minutes with a caput and that does work and gives the correct `drvl`.

Thoughts on this? @AlexanderWells-diamond

Reply:
They are not quite the same test. One test checks that a PandA change is reflected into EPICS, and the other tests that an EPICS change is reflected into PandA. Observe that this deleted test has no `caput` calls - it's only watching for changes that come over the wire from the `*CHANGES?` requests. The other test uses a `caput` to trigger the opposite behaviour.

It is certainly possible to do both aspects in one test, but it seems cleaner to me to keep the separation.

As for the conversion, I'm afraid I'm finding it a little hard to follow the expected behaviour of the `standard_responses`. It looks like it creates the "DELAY" `TimeFieldInfo`, then sets its ".MIN" parameter to 1.3e-10 (the conversion from ms to minutes, but apparently without changing the units?), then changes the units to ms with an associated change in .MIN,  and then lastly changes the units to s but with no MIN change. It's hard to see what it's supposed to be.

I would suggest setting up a camonitor on the IOC, outside of the tests, and just observe what is definitely happening on the PV (and its various fields).


Also of note is that these tests did use a specialised `dummy_server_time`, rather than the generic one, as they needed slightly different data. Perhaps have another look back through that and see if there's a missing piece of data somewhere?

Reply:
Ahhh I figured it out, the `update_drvl` gets the `MIN` directly from the panda and uses that for the conversion

https://github.com/PandABlocks/PandABlocks-ioc/blob/9c6495ea39f93a1874524fd94020c53a56014909/src/pandablocks_ioc/ioc.py#L498

I'll just have to change the test, or fake panda responses a little.